hydromt/gis_utils.py

```diff
@@ -365,7 +364,7 @@ def affine_to_coords(transform, shape, x_dim="x", y_dim="y"):
     if not isinstance(transform, Affine):
         transform = Affine(*transform)
     height, width = shape
-    if transform.b == 0:
+    if transform.b == transform.d == 0:
```

Comment:
if `transform.b` or `transform.d` are floats and not an integers this is very error prone. In that case please use `np.isclose`